### PR TITLE
Add TODO Tools Hackathon Summary

### PR DESCRIPTION
**Pull request by caniszczyk**
A summary of the TODO Tools Hackathon for the blog.

Signed-off-by: Chris Aniszczyk <caniszczyk@gmail.com>

### CODE REVIEW

**Comment by caniszczyk** [comment]
Hey @jeffmcaffer and other TODO Group folks, RFC for the blog?

**Comment by jeffmcaffer** [comment]
I added some more points in the crawler discussion but am otherwise good to go.

File: _posts/2017-06-21-todo-tools-hackathon.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: "TODO Tools Hackathon June 2017"
+author: todogroup
+---
+
+We recently held a TODO Tools Working Group hackathon at Microsoft. There were about a dozen people from half a dozen different organizations. We ended up with three groupings of hacking around: repository linting, GitHub portal and GitHub data/crawling. Below is a summary of the work that was done and some next steps.
+
+### Repolinter ([https://github.com/todogroup/repolinter](https://github.com/todogroup/repolinter))
+
+Repolinters is a a simple linter to check for open source quality: [https://github.com/todogroup/repolinter](https://github.com/todogroup/repolinter). We initially evaluated using [https://github.com/basicallydan/forkability](https://github.com/basicallydan/forkability) but it wasn’t designed for extensibility and isn’t being actively maintained.
```
Comment (trevmex) [inline _posts/2017-06-21-todo-tools-hackathon.md]: Repolinters should be Repo Linter

File: _posts/2017-06-21-todo-tools-hackathon.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+title: "TODO Tools Hackathon June 2017"
+author: todogroup
+---
+
+We recently held a TODO Tools Working Group hackathon at Microsoft. There were about a dozen people from half a dozen different organizations. We ended up with three groupings of hacking around: repository linting, GitHub portal and GitHub data/crawling. Below is a summary of the work that was done and some next steps.
+
+### Repolinter ([https://github.com/todogroup/repolinter](https://github.com/todogroup/repolinter))
```
Comment (trevmex) [inline _posts/2017-06-21-todo-tools-hackathon.md]: This should be Repo Linter

File: _posts/2017-06-21-todo-tools-hackathon.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Rules:
+
+* Most rules boiled down to file existence (e.g. README, LICENSE, etc.) or file contents (e.g. source files contain license header).
+
+* Language-specific rules (requires linguist) such as verifying that NodeJS projects have a package.json file.
+
+Next steps:
+
+* Move the rule configuration outside of the tool source code so teams can define their own configuration.
```
Comment (trevmex) [inline _posts/2017-06-21-todo-tools-hackathon.md]: This is done.

File: _posts/2017-06-21-todo-tools-hackathon.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+* Performance improvements (especially in file_starts_with rule).
+
+* More structure around rule config and results (e.g. provide advice on how to fix, warnings vs errors vs suggestions).
+
+* Support running against a remote GitHub repository.
```
Comment (trevmex) [inline _posts/2017-06-21-todo-tools-hackathon.md]: This is done.

File: _posts/2017-06-21-todo-tools-hackathon.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+* Most rules boiled down to file existence (e.g. README, LICENSE, etc.) or file contents (e.g. source files contain license header).
+
+* Language-specific rules (requires linguist) such as verifying that NodeJS projects have a package.json file.
+
+Next steps:
+
+* Move the rule configuration outside of the tool source code so teams can define their own configuration.
+
+* Performance improvements (especially in file_starts_with rule).
```
Comment (trevmex) [inline _posts/2017-06-21-todo-tools-hackathon.md]: The file_starts_with performance improvements have been done.

Comment (caniszczyk) [inline _posts/2017-06-21-todo-tools-hackathon.md]: fixed

File: _posts/2017-06-21-todo-tools-hackathon.md (hunk 4 above)
Comment (caniszczyk) [inline _posts/2017-06-21-todo-tools-hackathon.md]: fixed

**Pull request merged**